vllm/_custom_ops.py

```diff
@@ -1272,11 +1272,12 @@ def reshape_and_cache_flash(
     kv_cache_dtype: str,
     k_scale: torch.Tensor,
     v_scale: torch.Tensor,
+    is_NHD: bool = True,
 ) -> None:
     torch.ops._C_cache_ops.reshape_and_cache_flash(key, value, key_cache,
                                                    value_cache, slot_mapping,
                                                    kv_cache_dtype, k_scale,
-                                                   v_scale)
+                                                   v_scale, is_NHD)
```

Comment:
Could we call `torch.permute(kv_cache, (0, 2, 1, 3))` at some higher level when the NHD layout is needed?

The NHD vs HND layouts can be implicitly captured in the strides of the `torch.Tensor`s